### PR TITLE
Fix Cmake files for windows compilation

### PR DESCRIPTION
**Pull request by aahmed-se**
Miscellaneous fixes for compilation on windows platforms.

### CODE REVIEW

File: pulsar-client-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -216,6 +216,7 @@ set(COMMON_LIBS
   ${COMMON_LIBS}
   ${Boost_REGEX_LIBRARY}
   ${Boost_SYSTEM_LIBRARY}
+  ${Boost_DATE_TIME_LIBRARY}
```
Comment (merlimat) [inline pulsar-client-cpp/CMakeLists.txt]: Why do we need date_time? We shouldn't be using it.

Comment (aahmed-se) [reply pulsar-client-cpp/CMakeLists.txt]: we use in the following places
```
./lib/stats/ProducerStatsImpl.h:#include <boost/date_time/local_time/local_time.hpp>
./lib/stats/ConsumerStatsBase.h:#include <boost/date_time/posix_time/posix_time.hpp>
./lib/stats/ProducerStatsBase.h:#include <boost/date_time/posix_time/posix_time.hpp>
./lib/BrokerConsumerStatsImpl.h:#include <boost/date_time/microsec_time_clock.hpp>
./lib/BrokerConsumerStatsImplBase.h:#include <boost/date_time/posix_time/ptime.hpp>
```

**Comment by aahmed-se** [comment]
run integration tests

**Comment by aahmed-se** [comment]
run integration tests

**Comment by aahmed-se** [comment]
run cpp tests

**Comment by aahmed-se** [comment]
run integration tests

**Comment by aahmed-se** [comment]
run integration tests

File: pulsar-client-cpp/CMakeLists.txt [hunk 2]
```diff
@@ -212,16 +216,30 @@ include_directories(
...
+    ${COMMON_LIBS}
+    ${Boost_REGEX_LIBRARY}
+    ${Boost_SYSTEM_LIBRARY}
+    ${CURL_LIBRARIES}
+    ${OPENSSL_LIBRARIES}
+    ${ZLIB_LIBRARIES}
+    ${Protobuf_LITE_LIBRARIES}
+    ${ADDITIONAL_LIBRARIES}
+  )
+endif()
```
Comment (merlimat) [inline pulsar-client-cpp/CMakeLists.txt]: Why not leave the original set non duplicated and add later: 

```
if (MSVC)
  set(COMMON_LIBS
        ${COMMON_LIBS}
        ${Boost_DATE_TIME_LIBRARY}
   )
endif ()	    
```

Comment (aahmed-se) [reply pulsar-client-cpp/CMakeLists.txt]: done

**Comment by aahmed-se** [comment]
run java8 tests

**Comment by aahmed-se** [comment]
run integration tests

**Pull request merged**